api/lib/opentelemetry/trace/span.rb

```diff
@@ -6,42 +6,143 @@
 
 module OpenTelemetry
   module Trace
-    # Span is a mutable object storing information about the current operation execution.
+    # Span represents a single operation within a trace. Spans can be nested to
+    # form a trace tree. Often, a trace contains a root span that describes the
+    # end-to-end latency and, optionally, one or more sub-spans for its
+    # sub-operations.
+    #
+    # Once Span {Tracer#start_span is created} - Span operations can be used to
+    # add additional properties to it like attributes, links, events, name and
+    # resulting status. Span cannot be used to retrieve these properties. This
+    # prevents the mis-use of spans as an in-process information propagation
+    # mechanism.
+    #
+    # {Span} must be ended by calling {#finish}.
     class Span
+      # Retrieve the spans SpanContext
+      #
+      # The returned value may be used even after the Span is finished.
+      #
+      # @return [SpanContext]
       attr_reader :context
 
+      # Spans must be created using {Tracer}. This is for internal use only.
+      #
+      # @api private
       def initialize(span_context: nil)
         @context = span_context || SpanContext.new
       end
 
+      # Return the flag whether this span is recording events
+      #
+      # @return [Boolean] true if this Span is active and recording information
+      #   like events with the #add_event operation and attributes using
+      #   #set_attribute.
       def recording_events?
         false
       end
 
+      # Set attribute
+      #
+      # Note that the OpenTelemetry project
+      # {https://github.com/open-telemetry/opentelemetry-specification/blob/master/semantic-conventions.md
+      # documents} certain "standard attributes" that have prescribed semantic
+      # meanings.
+      #
+      # @param [String, Symbol] key
+      # @param [String, Symbol, Boolean, Numeric] value
+      #
+      # @return [self] returns itself
       def set_attribute(key, value)
         raise ArgumentError unless valid_key?(key)
         raise ArgumentError unless valid_value?(value)
+
+        self
```

Comment:
Indentation is wrong here.

Reply:
Are you sure, seems correct to me

Reply:
Strange - this rendered incorrectly on an iPad, but looks correct when viewed on my laptop 🤷‍♂